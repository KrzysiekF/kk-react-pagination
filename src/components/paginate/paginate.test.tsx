import React, { ReactElement } from 'react';
import { render } from '@testing-library/react';

import Paginate, { Props } from '.';

const generateElements = (count: number) => {
    const data: Array<ReactElement> = [];

    for (let i = 1; i <= count; i++) {
        data.push(<div key={i}>Element #{i}</div>);
    }

    return data;
};

describe('Paginate', () => {
    let props: Props;

    beforeEach(() => {
        props = {
            pageSize: 5,
            data: generateElements(10),
        };
    });

    describe('render()', () => {
        it('renders a submit button', () => {
            const { container } = render(<Paginate {...props} />);
            expect(container.firstChild).toMatchSnapshot(`
        <div class="krp">
            <div class="krp__list">
                <div>Element #1</div>
                <div>Element #2</div>
                <div>Element #3</div>
                <div>Element #4</div>
                <div>Element #5</div>
            </div>
            <div class="krp__pagination">
                <ul class="krp__pagination-list pagination-1-2-3">
                    <li><button>prev</button></li>
                    <li class="active-1-2-4"><button>1</button></li>
                    <li class=""><button>2</button></li>
                    <li><button>next</button></li>
                </ul>
            </div>
        </div>
      `);
        });
    });
});
