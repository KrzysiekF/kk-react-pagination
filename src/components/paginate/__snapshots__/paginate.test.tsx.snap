// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginate render() renders a submit button: 
        <div class="krp">
            <div class="krp__list">
                <div>Element #1</div>
                <div>Element #2</div>
                <div>Element #3</div>
                <div>Element #4</div>
                <div>Element #5</div>
            </div>
            <div class="krp__pagination">
                <ul class="krp__pagination-list pagination-1-2-3">
                    <li><button>prev</button></li>
                    <li class="active-1-2-4"><button>1</button></li>
                    <li class=""><button>2</button></li>
                    <li><button>next</button></li>
                </ul>
            </div>
        </div>
       1`] = `
<div
  class="krp"
>
  <div
    class="krp__list"
  >
    <div>
      Element #
      1
    </div>
    <div>
      Element #
      2
    </div>
    <div>
      Element #
      3
    </div>
    <div>
      Element #
      4
    </div>
    <div>
      Element #
      5
    </div>
  </div>
  <div
    class="krp__pagination"
  >
    <ul
      class="krp__pagination-list pagination-0-2-1"
    >
      <li>
        <button>
          prev
        </button>
      </li>
      <li
        class="active-0-2-2"
      >
        <button>
          1
        </button>
      </li>
      <li
        class=""
      >
        <button>
          2
        </button>
      </li>
      <li>
        <button>
          next
        </button>
      </li>
    </ul>
  </div>
</div>
`;
